### PR TITLE
Allow optional extra parameters to filter scope

### PR DESCRIPTION
**Pull request by orobogenius**
This PR allow optional extra to filter scope. See #3

### CODE REVIEW

File: src/BaseFilters.php [hunk 1]
```diff
@@ -28,12 +28,16 @@ public function __construct(Request $request)
      * corresponding to request query parameters.
      * 
      * @param \Illuminate\Database\Eloquent\Builder $builder
+     * @param string $extraFilters
```
Comment (mykeels) [inline src/BaseFilters.php]: Could you please make the type `array`?

File: src/BaseFilters.php [hunk 2]
```diff
@@ -28,12 +28,16 @@ public function __construct(Request $request)
...
      * @return \Illuminate\Database\Eloquent\Builder
     */
-    public function apply(Builder $builder):Builder
+    public function apply(Builder $builder, $extraFilters = null):Builder
     {
         $this->builder = $builder;
-        foreach ($this->filters() as $name => $value) {
+
+        $filters = $extraFilters ? array_merge($this->filters(), $extraFilters) : $this->filters();
```
Comment (mykeels) [inline src/BaseFilters.php:38]: We might want to confirm the type of `$extraFilters` with `is_array(...)` before making it participate in the merge, so we don't run into a type error.

In the line before this, we could have:

```php
if ($extraFilters && !is_array($extraFilters))
{
  throw new \Exception('$extraFilters must be of array type");
}
```

or something similar.

File: src/Traits/FilterableTrait.php [hunk 3]
```diff
@@ -12,10 +12,11 @@
      * 
      * @param \Illuminate\Database\Eloquent\Builder $query
      * @param \Mykeels\Filters\BaseFilters $filters
+     * @param string $extraFilters
```
Comment (mykeels) [inline src/Traits/FilterableTrait.php]: Could you please make the type `array`?

**Review by mykeels — CHANGES_REQUESTED** [review]
Please reply or make the following changes.

Also, were you able to run the linter and static-analysis tool?

If not, run the following in the terminal:

```
composer run lint
composer run analyse
```

If the linter has issues, you might want to fix them with `composer run fix-lint`

File: src/BaseFilters.php (hunk 1 above)
Comment (orobogenius) [reply src/BaseFilters.php]: Oh, wow, it was meant to be an array, my bad.

File: src/BaseFilters.php (hunk 2 above)
Comment (orobogenius) [reply src/BaseFilters.php:38]: We might type the `$extraFilters` parameter as array in the argument list.

Comment (mykeels) [reply src/BaseFilters.php:38]: Okay, you could try that, but I'm thinking we might have an exception thrown if the user does not provide the parameter's value.

Comment (orobogenius) [reply src/BaseFilters.php:38]: The parameter is meant to be optional, they might not provide a value for the parameter but if they do, it must be an array as would be type-hinted by the method's signature.

**Comment by orobogenius** [comment]
Fix the linter issues @mykeels, and made the requested changes. The `filter(...)` scope and `apply(...)` method would now naturally throw an exception if a type other than array is passed.

**Review by mykeels — APPROVED** [review]

**Pull request merged**